source/common/router/rds_impl.cc

```diff
@@ -241,6 +257,37 @@ void RdsRouteConfigProviderImpl::onConfigUpdate() {
     prev_config->config_ = new_config;
     return previous;
   });
+
+  const auto aliases = config_update_info_->resourceIdsInLastVhdsUpdate();
+  // Regular (non-VHDS) RDS updates don't populate aliases fields in resources.
```

Comment:
There's RouteConfigUpdateReceiver which is responsible for keeping and updating of RouteConfig state; it handles RDS and VHDS updates via onRdsUpdate and onVhdsUpdate respectively.
RdsRouteConfigProviderImpl doesn't contain much vhds-specific logic: on a VHDS update it will also call callbacks notifying ActiveConnections of the update. 

I felt that maintaining RouteConfig state is where the protocol-specific logic belongs to; otoh the amount of VHDS-specific logic is so minimal in the RdsRouteConfigProviderImp I felt that creating a dedicated provider is not worth it. Another way to look at it is a VHDS subscription is a child of an RDS subscription and they both contribute to the same RouteConfig. Splitting providers into RDS- and VHDS- specific ones is possible, but: their implementation would be very similar, their runtime states would be very similar. There would need to be some sort of a proxy provider in front of them to select a provider to serve a given call. IIRC I originally went with an implementation similar to what I described, but discarded it in favour of the current one.